src/seq_analysis/orf.rs

```diff
@@ -0,0 +1,152 @@
+// Copyright 2014-2016 Johannes Köster, Martin Larralde.
+// Licensed under the MIT license (http://opensource.org/licenses/MIT)
+// This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! One-way orf finder algorithm.
+//! complexity: O(n).
+
+//! # Example
+//!
+//! ```
+//! use bio::seq_analysis::orf::NaiveFinder;
+//! let start_codons = vec!(b"ATG");
+//! let stop_codons  = vec!(b"TGA", b"TAG", b"TAA");
+//! let min_len:usize = 50;
+//! let finder = NaiveFinder::new(start_codons, stop_codons, min_len);
+//!
+//! let sequence = b"ACGGCTAGAAAAGGCTAGAAAA"
+//!
+//! for orf in finder.find_all(sequence) {
+//!    ...do something...
+//! }
+//! ```
+
+//! Right now the only way to check the reverse strand for orf is to use
+//! the alphabet::dna::RevComp struct and to check for both sequences.
+//! But that's not so performance friendly, as the reverse complementation and the orf research
+//! could go on at the same time.
+
+
+/// Finder algorithm
+pub struct NaiveFinder <'a> {
+    start_codons: Vec<&'a [u8; 3]>,
+    stop_codons: Vec<&'a [u8; 3]>,
+    min_len: usize,
+}
+
+impl<'a> NaiveFinder <'a> {
+    /// Create a new instance for given genetic datas
+    pub fn new (start_codons: Vec<&'a [u8; 3]>, stop_codons: Vec<&'a [u8; 3]>, min_len: usize) -> Self {
+        NaiveFinder {
+            start_codons: start_codons,
+            stop_codons: stop_codons,
+            min_len: min_len,
+        }
+    }
+    /// Find all open reading frames in given sequence. Matches are returned as iterators over vectors
+    pub fn find_all(&'a self, sequence: &'a [u8]) -> Matches {
```

Comment:
Can you use the new ``TextSlice`` type alias here (see bio::utils)? Of course, since your algorithm is O(n) it would be even better to use ``IntoTextIterator``, but only if the code does not become over complicated.